server/publications/collections/product.js

```diff
@@ -1,7 +1,47 @@
-import { Products, Revisions } from "/lib/collections";
+import { Media, Products, Revisions } from "/lib/collections";
 import { Logger, Reaction } from "/server/api";
 import { RevisionApi } from "/imports/plugins/core/revisions/lib/api/revisions";
 
+export function findProductMedia(publicationInstance, productIds) {
+  const shopId = Reaction.getShopId();
+  const selector = {};
+
+  if (!shopId) {
+    return publicationInstance.ready();
+  }
+
+  if (Array.isArray(productIds)) {
+    selector["metadata.productId"] = {
+      $in: productIds
+    };
+  } else {
+    selector["metadata.productId"] = productIds;
+  }
+
+  if (shopId) {
+    selector["metadata.shopId"] = shopId;
+  }
+
+  // Product editors can see both published and unpublished images
+  if (!Reaction.hasPermission(["createProduct"], publicationInstance.userId)) {
+    selector["metadata.workflow"] = {
+      $in: [null, "published"]
+    };
+  } else {
```

Comment:
I dare not question the zenweseal's masterpiece https://github.com/reactioncommerce/reaction/blame/1815-media-take2/server/publications/collections/media.js#L30

But on a serious note, I suppose it makes sense, but not sure if you built it that way for a reason.